src/jv.c

```diff
@@ -1344,8 +1344,13 @@ int jv_contains(jv a, jv b) {
   } else if (jv_get_kind(a) == JV_KIND_ARRAY) {
     r = jv_array_contains(jv_copy(a), jv_copy(b));
   } else if (jv_get_kind(a) == JV_KIND_STRING) {
-    r = _jq_memmem(jv_string_value(a), jv_string_length_bytes(jv_copy(a)),
```

Comment:
Is it also the case that a haystack length of zero has undefined behavior anywhere?

And, what should `""|contains("")` return?

Reply:
It might be, actually.
My mac says this:
```If big_len is smaller than little_len, if little_len is 0, if big_len is 0 or if little
     occurs nowhere in big, NULL is returned; otherwise a pointer to the first character of
     the first occurrence of little is returned.```

Other systems say this:
```
The memmem() function returns a pointer to the beginning of the
       substring, or NULL if the substring is not found.

BUGS
In glibc 2.0, if needle is empty, memmem() returns a pointer to the
       last byte of haystack.  This is fixed in glibc 2.1.
```

So, it's not consistent anywhere. What behavior do we want a haystack of length zero to have? And what do we want `""|contains("")` to return?

Reply:
I think `"" | contains("")` can be true, false, null, an error -- undefined behavior, if you wish -- but the user-friendly thing to do is to say that it outputs `true`, IMO, and, for that matter, `"foobar" | contains("")` should be `true` as well, no?, after all, any string contains the empty string in a sense.

Reply:
Let's deal with the zero-length haystack question separately.